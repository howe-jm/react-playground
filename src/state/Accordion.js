import React from 'react';

export default class Accordion extends React.Component {
  static defaultProps = {
    sections: [],
  };

  state = { activeState: null };

  setActiveState = (i) => {
    if (this.state.activeState !== i) {
      this.setState({ activeState: i });
    } else {
      this.setState({ activeState: null });
    }
  };

  renderLiItems(section, i, activeState) {
    return (
      <li key={i}>
        <button type='button' onClick={() => this.setActiveState(i)}>
          {section.title}
        </button>
        {activeState === i && <p> {section.content} </p>}
      </li>
    );
  }

  render() {
    const { activeState } = this.state;
    const { sections } = this.props;
    return <ul>{sections.map((section, i) => this.renderLiItems(section, i, activeState))}</ul>;
  }
}
