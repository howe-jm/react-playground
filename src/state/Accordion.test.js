import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
];

describe(`Accordion component`, () => {
  it('Renders no sections by default', () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Renders when given an empty array', () => {
    const wrapper = shallow(<Accordion sections={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Renders when given no props', () => {
    const wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Responds to button clicks', () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Only opens the most recent section clicked', () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    wrapper.find('button').at(1).simulate('click');
    wrapper.find('button').at(0).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
