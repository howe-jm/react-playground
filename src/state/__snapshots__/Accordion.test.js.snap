// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component Only opens the most recent section clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
    <p>
       
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
       
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component Renders no sections by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component Renders when given an empty array 1`] = `<ul />`;

exports[`Accordion component Renders when given no props 1`] = `<ul />`;

exports[`Accordion component Responds to button clicks 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
    <p>
       
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
       
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;
